lib/jwt/decode.rb

```diff
@@ -33,18 +34,34 @@ def decode_segments
     private
 
     def verify_signature
+      return unless @key || @verify
+
+      return if none_algorithm?
+
+      raise JWT::DecodeError, 'No verification key available' unless @key
+
+      Array(@key).each do |key|
+        # return nil to avoid rubocop raising Lint/NonLocalExitFromIterator
```

Comment:
Can this RuboCop issue be avoided by using something like:
```
return if Array(@key).any?{ |key| verify_signature_for?(key) }
```

Reply:
great suggestion! done.